core/cloudflow-runner/src/main/scala/cloudflow/runner/Runner.scala

```diff
@@ -68,17 +67,20 @@ object Runner extends RunnerConfigResolver with StreamletLoader {
         val streamletExecution = loadedStreamlet.streamlet.run(withPodRuntimeConfig)
         loadedStreamlet.streamlet.logStartRunnerMessage(formatBuildInfo)
 
-        Try {
-          // the runner waits for the execution to complete
-          // In normal circumstances it will run forever for streaming data source unless
-          // being stopped forcibly or any of the queries faces an exception
-          Await.result(streamletExecution.completed, Duration.Inf)
-        } match {
-          case Success(_) ⇒ System.exit(0)
-          case Failure(ex @ ExceptionAcc(exceptions)) ⇒
+        // the runner waits for the execution to complete
+        // In normal circumstances it will run forever for streaming data source unless
+        // being stopped forcibly or any of the queries faces an exception
+        Await.ready(streamletExecution.completed, Duration.Inf)
+        streamletExecution.completed.value match {
```

Comment:
If you use `Await.result` you don't need this line?

Reply:
If you use `result`, you will throw an exception when it terminates due to a failure. You'd need an additional `try...catch` like it was done before. Like this, we have a uniform result handling.

Reply:
Yes but in this way you are handling a case that will never happen otherwise (the None case).
Just feels weird to do a `Future.value`, (I never use it) and do this in two steps. Remove line 74 and everything still compiles.. Just feels wrong. my two cents.